common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/RemoteBlockPushResolver.java

```diff
@@ -470,6 +527,39 @@ void closeAndDeleteOutdatedPartitions(
       });
   }
 
+  void deleteMergedFiles(AppAttemptShuffleMergeId appAttemptShuffleMergeId, int[] reduceIds) {
+    removeAppShufflePartitionInfoFromDB(appAttemptShuffleMergeId);
+    AppShuffleInfo appShuffleInfo = validateAndGetAppShuffleInfo(appAttemptShuffleMergeId.appId);
+    int shuffleId = appAttemptShuffleMergeId.shuffleId;
+    int shuffleMergeId = appAttemptShuffleMergeId.shuffleMergeId;
+    for (int reduceId : reduceIds) {
+      try {
+        File dataFile =
+            appShuffleInfo.getMergedShuffleDataFile(shuffleId, shuffleMergeId, reduceId);
+        dataFile.delete();
+      } catch (Exception e) {
```

Comment:
There is no `Exception` thrown during deletion (no security manager).
It returns a `boolean` indicating successful or failed delete.

We can keep track of that, and emit a consolidated log message at the end if `dataFileDeleteCount > 0 || indexFileDeleteCount > 0 || metaDeleteCount > 0`

Reply:
Could we log warn messages if `File.delete()` return false?

Reply:
This is a best effort deferred task, which can conflict with concurrent cleanup (like app termination for example) - so a number of failures is expected.
For busy clusters, a warn message will quickly overwhelm the logs.

Reply:
Add a consolidated log message at the end.

Reply:
Can we fix this @wankunde ?
Essentially, the changes are:

a) There is no exception thrown in this block - we dont need the try/catch.
b) When the delete fails, we dont need to do the `warn` (here and below) : this can happen if application exit is racing against remove shuffle - the `info` at the end of the method will suffice.